kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/handlers/CommonHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.admin.kafka.admin.handlers;
+
+import graphql.schema.DataFetchingEnvironment;
+import io.strimzi.admin.kafka.admin.AdminClientWrapper;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+
+public class CommonHandler {
```

Comment:
Just an observation, I would have dealt with the security differently. The `routingContext` is provided to keep a context for the request and be available from any handlers. We are also making the `routingContext` available in any DataFetcher. I would have created security handlers that deal with the different authentication mechanisms as we are going to support both SASL and OAuth2 and put them at the top of the handler list so that they get called first in the pipeline. The security handlers can create a "User" object on the `routingContext` allowing any other handler or DataFetcher to access it and use it to decorate the properties.